lib/edge/targeting.ts

```diff
@@ -23,6 +23,7 @@ type UserIdentifiers = {
 
 type TargetingRequest = {
   ids: string[];
+  hids?: string[];
```

Comment:
I'd suggest to make it mandatory since there is no semantic difference between empty and undefined in `Targeting` and it saves you from having to handle undefined there.

Also it consolidates normalization in `normalizeTargetingRequest`